src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -315,37 +319,49 @@ Status CoreWorkerDirectTaskSubmitter::KillTask(TaskSpecification task_spec,
 
           if (scheduled_tasks->second.empty()) {
             task_queues_.erase(scheduling_key);
+            CancelWorkerLeaseIfNeeded(scheduling_key);
           }
-          // Try to cancel lease requests
+          // Try to cancel the lease requests.
```

Comment:
I think this comment should be moved up.

Reply:
Let me remove this comment--it does not add much and was more of a reminder to myself.